src/cpp/r/RSexp.cpp

```diff
@@ -838,6 +838,77 @@ bool inherits(SEXP object, const char* S3Class)
    return Rf_inherits(object, S3Class);
 }
 
+std::set<std::string> makeNseFunctions()
+{
+   std::set<std::string> nseFunctions;
+   nseFunctions.insert("quote");
+   nseFunctions.insert("substitute");
+   nseFunctions.insert("match.call");
+   nseFunctions.insert("library");
+   nseFunctions.insert("require");
+   nseFunctions.insert("enquote");
+   nseFunctions.insert("expression");
+   nseFunctions.insert("evalq");
+   nseFunctions.insert("subset");
+   return nseFunctions;
+}
+
+std::set<std::string> nseFunctions()
+{
+   static const std::set<std::string> set = makeNseFunctions();
+   return set;
+}
+
+bool maybePerformsNSEImpl(SEXP node)
+{
+   if (Rf_isExpression(node))
```

Comment:
Hmm, I was under the impression that function bodies were expression objects, but I guess I'm wrong. Although, someone could still manually construct an expression object with `as.expression()`, but I guess those objects aren't actually seen within functions / callable objects.